modules/ngap_module/NgapApi.cc

```diff
@@ -434,6 +434,11 @@ string NgapApi::convert_ngap_resty_path_to_data_access_url(const std::string &re
     cmr_response.Parse(cmr_json_string.c_str());
     data_access_url = find_get_data_url_in_granules_umm_json_v1_4(restified_path, cmr_response);
 
+    // Check for existing .dmrpp and remove it if found. - kln 6/6/25
+    size_t pos = data_access_url.find(".dmrpp");
```

Comment:
This will avoid replacing the .dmrpp in the middle of the string. I think the code should not do that. Admittedly, that seems like an obscure case, but it might happen.

const std::string suffix = ".dmrpp";

if (data_access_url.size() >= suffix.size() &&
    str.compare(data_access_url.size() - suffix.size(), suffix.size(), suffix) == 0) {
    str.erase(data_access_url.size() - suffix.size());
}